api4/post_test.go

```diff
@@ -602,25 +602,32 @@ func TestCreatePostCheckOnlineStatus(t *testing.T) {
 	wsClient, err := th.CreateWebSocketClientWithClient(cli)
 	require.Nil(t, err)
 	defer func() {
-		wsClient.Close()
 		wg.Wait()
-
+		wsClient.Close()
 	}()
 
 	wsClient.Listen()
 
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
+		timeout := time.After(2 * time.Second)
+		cnt := true
 		i := 0
-		for ev := range wsClient.EventChannel {
-			if ev.EventType() == model.WEBSOCKET_EVENT_POSTED {
-				if i == 0 {
-					assert.False(t, ev.GetData()["set_online"].(bool))
-				} else {
-					assert.True(t, ev.GetData()["set_online"].(bool))
+		for cnt {
+			select {
+			case ev := <-wsClient.EventChannel:
+				if ev.EventType() == model.WEBSOCKET_EVENT_POSTED {
+					if i == 0 {
+						assert.False(t, ev.GetData()["set_online"].(bool))
+					} else {
+						assert.True(t, ev.GetData()["set_online"].(bool))
+					}
+					i++
 				}
-				i++
+				cnt = i != 2
+			case <-timeout:
+				cnt = false
```

Comment:
Interesting way to approach this. I would have used a `break`  statement on a timeout or when `i` becomes 2. :)

Reply:
`break` breaks the `select` we need to break `for` loop. This would require a named loop but this way it looked cooler 😛